src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -90,16 +90,16 @@ SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
 	SGVector<float64_t> vec(feats->get_dim_feature_space()+1);
 	switch (kernel)
 	{
-		case GAUSSIAN:
+		case GAUSSIAN:		
+			#pragma omp parallel for
```

Comment:
Is the result deterministic? Do you get the same result with the same seed?

Reply:
Also you might observe that there are repeating patterns in the matrix because each thread would start from the same state.. I know that can happen with rand...

Reply:
http://www.thesalmons.org/john/random123/papers/random123sc11.pdf You can see it’s a bit more difficult than that... For a nice explanation on how to achieve this https://github.com/google/jax/blob/master/design_notes/prng.md

Reply:
I suppose you're talking about [this](https://stackoverflow.com/questions/4287531/how-to-generate-random-numbers-in-parallel)?

Reply:
for example, yes. The point is that first you have no guarantee that omp will schedule the threads in the same way for different runs, then there is the issue with dispatching the state... long story short, this is something we will have to work on at some point, but I don't think what you wrote does what you expect it to. This is also beyond the scope of this PR imo